PurchasesCoreSwift/Purchasing/PurchasesOrchestrator.swift

```diff
@@ -192,9 +189,12 @@ public typealias RCDeferredPromotionalPurchaseBlock = (@escaping PurchaseComplet
         productsManager.cacheProduct(product)
 
         lock.lock()
+        defer {
```

Comment:
@beylmk @vegaro in case y'all haven't seen this, it's really neat. https://www.hackingwithswift.com/new-syntax-swift-2-defer

Reply:
so cool!